metaflow/plugins/logs_cli.py

```diff
@@ -0,0 +1,306 @@
+from metaflow._vendor import click
+from metaflow.cli import LOGGER_TIMESTAMP
+
+from ..exception import CommandException
+from ..datastore import TaskDataStoreSet, TaskDataStore
+
+
+from ..mflog import mflog, LOG_SOURCES
+
+# main motivation from https://github.com/pallets/click/issues/430
+# in order to support a default command being called for a Click group.
+#
+# NOTE: We need this in order to not introduce breaking changes to existing CLI, as we wanted to
+# nest both existing `logs` and the new `logs scrub` under a shared group, but `logs` already has
+# a well defined behavior of showing the logs.
+class CustomGroup(click.Group):
```

Comment:
we need to check if this is handled by @madhur-ob 's click parsing thing as well (should be but mentioning).

Reply:
tested, and this does not work as-is with the low level click api due to the fallback logic happening when invoking the command. Calling the explicit subcommand works as expected though, but the group level fallback fails

```
from metaflow.metaflow_runner import MetaflowAPI
from metaflow.cli import start

if __name__ == "__main__":

    api = MetaflowAPI.from_cli("AttemptLog.py", start)

    # works
    cmd = api().logs().show(input_path="217190/start")
    print(cmd)

    # does not
    cmd = api().logs(input_path="217190/start")
    print(cmd)
```

Will look into how easy of a fix there could be to support it

Reply:
Cool -- we don't need to fix it for this but ya, we need to probbaly have it working at some point. As in, the low level api should be able to handle all our cases.